src/test/util_tests.cpp

```diff
@@ -463,14 +566,26 @@ BOOST_AUTO_TEST_CASE(util_GetChainName)
     test_args.ParseParameters(0, (char**)argv_testnet);
     test_args.ReadConfigString(testnetconf);
     BOOST_CHECK_EQUAL(test_args.GetChainName(), "test");
+    // check setting the ConfigSection to test (and thus making
+    // [test] regtest=1 potentially relevent) doesn't break things
+    test_args.SelectConfigSection("test");
+    BOOST_CHECK_EQUAL(test_args.GetChainName(), "test");
 
     test_args.ParseParameters(2, (char**)argv_testnet);
     test_args.ReadConfigString(testnetconf);
     BOOST_CHECK_EQUAL(test_args.GetChainName(), "test");
+    test_args.SelectConfigSection("test");
+    BOOST_CHECK_EQUAL(test_args.GetChainName(), "test");
 
-    test_args.ParseParameters(2, (char**)argv_regtest);
+    test_args.ParseParameters(2, (char**)argv_testnet);
     test_args.ReadConfigString(testnetconf);
-    BOOST_CHECK_THROW(test_args.GetChainName(), std::runtime_error);
```

Comment:
You've removed this test that `GetChainName()` will throw if both `regtest` and `testnet` have been set. Is that intentional?

Reply:
No, I think that was a mistake while rebasing.